src/utils/TagUtils.ts

```diff
@@ -66,8 +70,8 @@ export default class TagUtils {
             }
 
             return categoriesCopy.map(category => {
-                const existingTagValue = existingPlayerTags[category.tag];
-                category.checked = !!existingTagValue;
```

Comment:
I don't see the reason for this rewrite. `category.checked = !!existingTagValue;` is even safer because it will always produce boolean. and it will handle undefined correctly too.

Reply:
same as above, think about the case where there aren't any remote tags yet. we want to default the checkbox to true. with the previous implementation, it would default to false

Reply:
got it. However it's not obvious from the code. How about a helper like this:
```
static getCheckedStatusForTagValue(tagValue: boolean | undefined): boolean {
        // If user does not have tag assigned to them, consider it selected
        if (tagValue === undefined) {
            return true;
        }

        return tagValue;
    }
```
It gives more room for explaining the logic and easy central place to change the logic in the future.

Reply:
Done.